components/teams-service/storage/postgres/postgres.go

```diff
@@ -179,29 +193,65 @@ func (p *postgres) EditTeam(ctx context.Context, team storage.Team) (storage.Tea
 
 // EditTeamByName edits the team's description and projects by name
 func (p *postgres) EditTeamByName(ctx context.Context,
-	teamName string, teamDescription string, teamProjects []string) (storage.Team, error) {
+	teamName string, teamDescription string, newProjects []string) (storage.Team, error) {
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	if newProjects == nil {
+		newProjects = []string{}
+	}
+
+	tx, err := p.db.BeginTx(ctx, nil)
+	if err != nil {
+		return storage.Team{}, p.processError(err)
+	}
 
 	projectsFilter, err := ProjectsListFromContext(ctx)
 	if err != nil {
 		return storage.Team{}, p.processError(err)
 	}
 
-	var t storage.Team
-	if teamProjects == nil {
-		teamProjects = []string{}
+	var oldProjects []string
+	err = tx.QueryRowContext(ctx,
+		`SELECT t.projects FROM teams t
```

Comment:
[nit] There's only one table involved, so we could drop the `t` from `teams t` and  `t.` from the columns.